dataprocess/dataprocess.go

```diff
@@ -0,0 +1,73 @@
+// Package dataprocess manages code related to the data-accepting process
+package dataprocess
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"image"
+	"os"
+
+	"github.com/pkg/errors"
+	pc "go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/rimage"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+// convertImageToPNG only supports rimage.LazyPNG and image.YCbCr.
+// This was done because decoding a png from the camera server and then re-encoding
+// it results in changes to the image that caused orbslam to be unable to use it
+// for feature matching (RSDK-1393).
+func convertImageToPNG(ctx context.Context, img image.Image) ([]byte, error) {
+	if lazyImg, ok := img.(*rimage.LazyEncodedImage); ok {
+		if lazyImg.MIMEType() != rdkutils.MimeTypePNG {
+			return nil, errors.Errorf("expected mime type %v, got %T", rdkutils.MimeTypePNG, img)
```

Comment:
We should have a test for this case

Reply:
Unfortunately, after all of this, Kat and I decided that this is the wrong place to be doing this image conversion. All of this code was saved but it will be going in a different PR and in a different file.